test/e2e/kubebuildertest.go

```diff
@@ -70,10 +71,50 @@ func (kt *kubebuilderTest) RunKubectlCommandWithInput(stdinInput string, cmdOpti
 	return string(output), err
 }
 
-// RunKubectlGetPodsInNamespace is a func to run kubectl get pods -n <namepsace> commands
-func (kt *kubebuilderTest) RunKubectlGetPodsInNamespace(testSuffix string, cmdOptions ...string) (string, error) {
-	getPodsOptions := []string{"get", "pods", "-n", fmt.Sprintf("e2e-%s-system", testSuffix)}
-	return kt.RunKubectlCommand(append(getPodsOptions, cmdOptions...)...)
+// RunKubectlInNamespace is a general func to run kubectl commands in the namespace
+func (kt *kubebuilderTest) RunKubectlInNamespace(cmdOptions ...string) (string, error) {
+	return kt.RunKubectlCommand(append([]string{"-n", kt.Namespace}, cmdOptions...)...)
+}
+
+// RunKubectlApply is a general func to run kubectl apply commands
+func (kt *kubebuilderTest) RunKubectlApply(inNamespace bool, cmdOptions ...string) (string, error) {
+	ops := append([]string{"apply"}, cmdOptions...)
+	if inNamespace {
+		return kt.RunKubectlInNamespace(ops...)
+	} else {
+		return kt.RunKubectlCommand(ops...)
+	}
+}
+
+// RunKubectlGet is a func to run kubectl get commands
+func (kt *kubebuilderTest) RunKubectlGet(inNamespace bool, cmdOptions ...string) (string, error) {
+	ops := append([]string{"get"}, cmdOptions...)
+	if inNamespace {
+		return kt.RunKubectlInNamespace(ops...)
+	} else {
+		return kt.RunKubectlCommand(ops...)
+	}
+}
+
+// RunKubectlDelete is a func to run kubectl delete commands
+func (kt *kubebuilderTest) RunKubectlDelete(inNamespace bool, cmdOptions ...string) (string, error) {
+	ops := append([]string{"delete"}, cmdOptions...)
+	if inNamespace {
+		return kt.RunKubectlInNamespace(ops...)
+	} else {
+		return kt.RunKubectlCommand(ops...)
+	}
+}
+
+// RunKubectlLogs is a func to run kubectl logs commands
+func (kt *kubebuilderTest) RunKubectlLogs(cmdOptions ...string) (string, error) {
+	ops := append([]string{"logs"}, cmdOptions...)
+	return kt.RunKubectlInNamespace(ops...)
+}
+
+// RunKubectlGetPods is a func to run kubectl get pods -n <namepsace> commands
+func (kt *kubebuilderTest) RunKubectlGetPods(inNamespace bool, cmdOptions ...string) (string, error) {
```

Comment:
Looks like we can create a `kubectl` object with `Get, Delete, Apply, Delete` methods. Will be much cleaner I think.